### ISSUE TITLE
Bump Maplibre Version to 4

### ISSUE THREAD

**Issue by brandonfcohen1** [watergis/maplibre-gl-legend#26]
**Is your feature request related to a problem? Please describe.**
I am using Maplibre 4.x which causes issues with `maplibre-gl-legend`.

**Describe the solution you'd like**
I'd like to make `maplibre-gl-legend` compatible with 4.x

**Describe alternatives you've considered**
I can manually override types in my app, but I'd rather fix here.

**Additional context**
Let me know if you want to keep this using 3.x for reasons

**Comment by JinIgarashi** [comment]
@brandonfcohen1 Yes, we should upgrade maplibre to v4. Could you make a PR to upgrade?🙏

**Comment by brandonfcohen1** [comment]
having some trouble with the upgrade- will revisit when I have more time!
